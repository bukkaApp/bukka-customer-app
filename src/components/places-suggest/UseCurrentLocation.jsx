import React from 'react';
import SmallSpinner from '../spinners/SmallSpinner';
import LocationArrow from '../icons/LocationArrow';

import { useLocationContext } from '../../context/LocationContext';
import './usecurrentlocation.scss';

const UseCurrentLocation = () => {
  const { setCurrentLocation, loading } = useLocationContext();
  return (
    <div
      className="suggestion-geo-group input-group"
      onClick={setCurrentLocation}
      tabIndex={0}
      role="link"
    >
      <div className="input-group-prepend">
        <span className="input-group-text location-arrow spinner-loading">
          {loading ? <SmallSpinner /> : <LocationArrow />}
        </span>
      </div>
      <h4 className="suggestion text-center d-block">Use current location</h4>
    </div>
  );
};

export default UseCurrentLocation;

UseCurrentLocation.propTypes = {};
