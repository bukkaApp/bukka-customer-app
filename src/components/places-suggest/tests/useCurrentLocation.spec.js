import React from 'react';

import UseCurrentLocation from '../UseCurrentLocation';

describe('Use current location ccomponent', () => {
  const store = mockStore({});

  const props = {
    selectLocation: jest.fn()
  };
  const wrapper = render(
    <Provider store={store}>
      <UseCurrentLocation {...props} />
    </Provider>
  );

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
