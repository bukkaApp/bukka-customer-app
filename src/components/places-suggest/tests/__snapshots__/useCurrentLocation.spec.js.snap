// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Use current location ccomponent renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="suggestion-geo-group input-group"
        role="link"
        tabindex="0"
      >
        <div
          class="input-group-prepend"
        >
          <span
            class="input-group-text location-arrow spinner-loading"
          >
            <svg
              class="css-3n0dwa e13fwc0q0"
              height="18"
              viewBox="0 0 18 18"
              width="18"
            >
              <defs>
                <path
                  d="M20 28h9v9l9-18z"
                  id="icon-current-location_svg__a"
                />
              </defs>
              <use
                fill="#9c1c26"
                fill-rule="evenodd"
                transform="translate(-20 -19)"
                xlink:href="#icon-current-location_svg__a"
              />
            </svg>
          </span>
        </div>
        <h4
          class="suggestion text-center d-block"
        >
          Use current location
        </h4>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="suggestion-geo-group input-group"
      role="link"
      tabindex="0"
    >
      <div
        class="input-group-prepend"
      >
        <span
          class="input-group-text location-arrow spinner-loading"
        >
          <svg
            class="css-3n0dwa e13fwc0q0"
            height="18"
            viewBox="0 0 18 18"
            width="18"
          >
            <defs>
              <path
                d="M20 28h9v9l9-18z"
                id="icon-current-location_svg__a"
              />
            </defs>
            <use
              fill="#9c1c26"
              fill-rule="evenodd"
              transform="translate(-20 -19)"
              xlink:href="#icon-current-location_svg__a"
            />
          </svg>
        </span>
      </div>
      <h4
        class="suggestion text-center d-block"
      >
        Use current location
      </h4>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
